pkg/controller/jobframework/reconciler.go

```diff
@@ -458,17 +459,28 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, object clie
 func (r *JobReconciler) constructWorkload(ctx context.Context, job GenericJob, object client.Object) (*kueue.Workload, error) {
 	podSets := job.PodSets()
 
+	jobGVK := job.GetGVK()
+
 	wl := &kueue.Workload{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      GetWorkloadNameForOwnerWithGVK(object.GetName(), job.GetGVK()),
+			Name:      GetWorkloadNameForOwnerWithGVK(object.GetName(), jobGVK),
 			Namespace: object.GetNamespace(),
+			Labels:    map[string]string{},
 		},
 		Spec: kueue.WorkloadSpec{
 			PodSets:   resetMinCounts(podSets),
 			QueueName: QueueName(job),
 		},
 	}
 
+	if utf8.RuneCountInString(job.Object().GetName()) < 64 {
+		wl.Labels[controllerconsts.ParentNameLabel] = job.Object().GetName()
+	}
+
+	if utf8.RuneCountInString(jobGVK.GroupKind().String()) < 64 {
```

Comment:
A group is validated as a DNS1123Subdomain, for a total of 253 characters. If we can't have all of them, we should drop this for consistency.

https://github.com/kubernetes/kubernetes/blob/99190634ab252604a4496882912ac328542d649d/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L293